File: DayTeller.cpp
#include<iostream.h>
#include<stdlib.h>
#include<conio.h>
int main()
{
const char *ad_1[]={"Sunday","Monday","Tuesday","Wednesday","Thursday",
                    "Friday","Saturday",};
char *tmp;
long int year;
int month,day,check[]={31,0,31,30,31,30,31,31,30,31,30,31},i;
do
{
clrscr();
do
{
cout<<"Enter the year (>=1) : ";
cin>>tmp;
year=atol(tmp);
}while(year<1);
if((year%4==0 && year%100!=0) || year%400==0)
{
check[1]=29;
}
else
{
check[1]=28;
}
do
{
cout<<"\nEnter the month (>=1 && <=12) : ";
cin>>tmp;
month=atoi(tmp);
}while(month<1 || month>12);
do
{
cout<<"\nEnter the day (>=1 && <="<<check[month-1]<<") : ";
cin>>tmp;
day=atoi(tmp);
}while(day<1 || day>check[month-1]);
cout<<"\nThe day "<<day<<"/"<<month<<"/"<<year<<" is a ";
year+=day-1+((year-1)/4)-((year-4)/100)+((year-1)/400);
for(i=month-2;i>=0;i--)
{
year+=check[i];
}
cout<<ad_1[year%7]<<".";
gotoxy(32,25);
cout<<"Press escape to exit, any other key to continue.";
}while(getch()!=27);
return(0);
}

File: DayTeller.java
/**
 *
 * @author VolitionalGod
 */
 
import java.util.Scanner;

public class DayTeller {

    private int day,month,i,check[];
    private long year,index;
    private String ad1[];
    Scanner s = new Scanner(System.in);
    
    DayTeller()
    {
        day=month=i=0;
        year=index=0;
        ad1 = new String[]{"Sunday","Monday","Tuesday","Wednesday","Thursday","Friday","Saturday"};
        check = new int[]{31,28,31,30,31,30,31,31,30,31,30,31};        
    }
        
    public void getYear()
    {
        do
        {
        System.out.print("Enter the year (>= 1) : ");
        year = s.nextLong();
        }while(year<=0);      
    }
    
    public void setLeap()
    {
        if(year%4==0 && year%100!=0 || year%400==0)
        {
            check[1]+=1;
        }
    }
    
    public void getMonth()
    {
        do
        {
        System.out.print("Enter the month (>= 1 AND <=12) : ");
        month = s.nextInt();
        }while(month<=0 || month>=13);        
    }
    
    public void getDay()           
    {
        do
        {
        System.out.print("Enter the day (>= 1 AND <="+check[month-1]+") : ");
        day = s.nextInt();
        }while(day<=0 || day>check[month-1]);        
    }
    
    public void genIndex()
    {
        index = day + (year-1) + ((year-1)/4) - ((year-1)/100) + ((year-1)/400);
        for(i=month-2;i>=0;i--)
        {
            index += check[i];
        }
        index %= 7;
    }
    
    public void displayDay()
    {
        System.out.println(day+"/"+month+"/"+year+" is a "+ad1[(int)index]+".");        
    }    
        
    
    public static void main(String[] args) {
        
        DayTeller obj = new DayTeller();
        obj.getYear();
        obj.setLeap();
        obj.getMonth();
        obj.getDay();
        obj.genIndex();
        obj.displayDay();                
    }
}
